symphony-ws/src/main/java/se/havochvatten/symphony/service/CalcService.java

```diff
@@ -836,5 +832,21 @@ public GridCoverage2D relativeDifference(GridCoverage2D base, GridCoverage2D sce
         var floatbase = operations.add(base, new double[]{0.0});
         return (GridCoverage2D) operations.divide(difference, floatbase);
     }
+
+    public List<CompoundComparisonSlice> getCompoundComparisons(Principal principal) {
```

Comment:
Why pass the whole principal object instead of the name?

Reply:
I've weighed both these alternatives and I think this is better with regard to "separation of concerns".
I think it's clearer that the application user be represented by a `Principal` object from the point of view of the calling class than e.g the string of characters in the username.
It shouldn't be the "concern" of the API class how exactly the coupling between the current user and 'their' associated entities is implemented exactly.

However, after looking at this snippet for a minute I think a better name for the parameter would be `user`, perhaps.

Reply:
Yes ok, i agree 👍